the-collab-lab/tcl-30-whats-near-me
# Create Placeholder Views and add Google Map to Map component

## Description

This PR adds  'Map', 'List' , Settings', 'Help' views as well as navigation for these views. It also ensures that a Google map with a hardcoded location and hardcoded zoom level is rendered in the Map view of the app.

## Related Issue

Closes [TCL30-6](https://the-collab-lab.atlassian.net/browse/TCL30-6) and [TCL30-7](https://the-collab-lab.atlassian.net/browse/TCL30-7)

## Acceptance Criteria

- react-router-dom has been added as a project dependency
- google-map-react has been added as a project dependency
- 4 navigation links are present at the bottom of the app (for the 'Map', 'List' , Settings', 'Help' views)
- When one of the links is clicked, the browser URL updates to represent the current view, and a matching view component is displayed ( Map → / ,  List → /list , Settings → /settings, Help → /help)
- Map component renders a default Google Map at a hardcoded lat/long & hardcoded zoom level in production, not development mode, using an API key


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|  ✓    | :link: Update dependencies |
|     | :scroll: Docs              |


### Before

![1](https://user-images.githubusercontent.com/16920088/125204409-d8c13300-e285-11eb-89dd-25633edcd760.png)

### After
![3](https://user-images.githubusercontent.com/16920088/125208011-b08eff80-e298-11eb-8cf7-a4028393b18f.png)

## Testing Steps / QA Criteria

- From your terminal, pull down this branch with ``git pull origin ku-am-placeholder-views`` and check out the branch  with ``git checkout ku-am-placeholder-views``
-  Run ``npm install`` to install the newly added  dependency and then ``npm start`` to launch the app. You should view the updates in your browser
- Make sure navbar is present at the bottom of the app
- Make sure a matching view is displayed when one of the links is selected
- Make sure  a Google Map with a hardcoded lat/long & hardcoded zoom level is rendered in the Map View 
- Make sure the map renders in production, not development mode, using an API key
